### PR TITLE
Add indicator arrow to LegendThreshold

### LINKED ISSUE #435: Implement current position arrow in legend threshold.
Enable our legend threshold to support a current position arrow. We previously implemented this in GCAT and it may be useful to add this feature in our packages.

Image

### PR DESCRIPTION
**Pull request by chasulin**
Closes #435 

Adds the option to have a current category indicator in LegendThreshold

[Storybook example (MetricLegendThreshold)](https://act-now-packages--pr481-casulin-arrow-on-leg-mrg19eut.web.app/storybook/index.html?path=/story/components-metriclegendthreshold--horizontal-with-indicator)

[Storybook example (LegendThreshold)](https://act-now-packages--pr481-casulin-arrow-on-leg-mrg19eut.web.app/storybook/index.html?path=/story/components-legendthreshold--horizontal-with-indicator)

`MetricLegendThreshold` now accepts an optional `CurrentCategory` prop. If this prop is passed, the corresponding category's `Item` in `Items[]` (passed into `LegendThreshold`) will have a `showIndicator` property set to `true`, and a little arrow will render above the midpoint of the category's section of the legend.

Screen Shot 2022-12-06 at 5 40 10 PM

### CODE REVIEW

File: packages/ui-components/src/components/LegendThreshold/LegendThresholdHorizontal.tsx [hunk 1]
```diff
@@ -29,44 +31,73 @@ export const LegendThresholdHorizontalInner = <T,>({
...
+                top={0}
+                left={(scaleRect(itemIndex) || 0) + rectWidth / 2 - 5}
+              >
+                <IndicatorPolygon points="0 0, 10 0, 5 7.5" />
+              </Group>
+            )
+          );
+        })}
+
+      <Group top={heightForIndicator}>
```
Comment (chasulin) [inline packages/ui-components/src/components/LegendThreshold/LegendThresholdHorizontal.tsx]: I wrapped the rectangles and the labels in a `Group`, which is shifted down [`heightForIndicator`px](https://github.com/covid-projections/act-now-packages/pull/481/files#diff-3921c12dc909ffb0a5014f0c67d810a34c28c8a0f0b82b6115e64a91a503f408R38-R40) (10px) if there is an indicator present, in order to make space

File: packages/ui-components/src/components/MetricLegendThreshold/MetricLegendThreshold.tsx [hunk 2]
```diff
@@ -3,54 +3,53 @@ import React from "react";
 import { Box, Stack, Typography } from "@mui/material";
 
 import { Metric } from "@actnowcoalition/metrics";
+import { Category } from "@actnowcoalition/metrics";
 
-import { LegendThreshold } from "../LegendThreshold";
+import { BaseLegendThresholdProps, LegendThreshold } from "../LegendThreshold";
```
Comment (chasulin) [inline packages/ui-components/src/components/MetricLegendThreshold/MetricLegendThreshold.tsx:8]: `LegendThreshold` and `MetricLegendThreshold` had some prop redundancies (`width`, `height`, `borderRadius`, `orientation`, `showLabels`), so i pulled them out into `BaseLegendThreshold`, which is now shared between the two components

File: packages/ui-components/src/components/MetricLegendThreshold/MetricLegendThreshold.tsx [hunk 3]
```diff
@@ -3,54 +3,53 @@ import React from "react";
...
   endLabel?: React.ReactNode;
   /** Whether or not to display metric name and info. If false, only thermometer is displayed. */
-  includeOverview?: boolean;
-  /** Border radius of the bars */
-  borderRadius?: number;
-  /** Width of the bars */
-  width?: number;
-  /** Height of the bars */
-  height?: number;
+  showOverview?: boolean;
```
Comment (chasulin) [inline packages/ui-components/src/components/MetricLegendThreshold/MetricLegendThreshold.tsx:21]: i renamed `includeOverview` to `showOverview`

**Comment by pnavarrc** [comment]
remember to add the changeset!

File: packages/ui-components/src/components/LegendThreshold/LegendThreshold.style.ts [hunk 4]
```diff
@@ -34,3 +34,9 @@ export const LegendColor = styled("div", { shouldForwardProp: isValidProp })<{
   ${(props) => getTopRadius(props.roundTop)}
   ${(props) => getBottomRadius(props.roundBottom)}
 `;
+
+export const IndicatorPolygon = styled("polygon")`
+  stroke: ${({ theme }) => theme.palette.common.black};
+  fill: ${({ theme }) => theme.palette.common.black};
+  stroke-width: 1px;
```
Comment (pnavarrc) [inline packages/ui-components/src/components/LegendThreshold/LegendThreshold.style.ts:41]: nit: you probably don't need the stroke if the polygon is filled

File: packages/ui-components/src/components/LegendThreshold/LegendThresholdHorizontal.tsx [hunk 5]
```diff
@@ -29,44 +31,73 @@ export const LegendThresholdHorizontalInner = <T,>({
...
+        items.map((item: T, itemIndex) => {
+          return (
+            getItemShowIndicator(item, itemIndex) && (
+              <Group
+                key={`indicator-item-${itemIndex}`}
+                top={0}
+                left={(scaleRect(itemIndex) || 0) + rectWidth / 2 - 5}
+              >
+                <IndicatorPolygon points="0 0, 10 0, 5 7.5" />
+              </Group>
```
Comment (pnavarrc) [inline packages/ui-components/src/components/LegendThreshold/LegendThresholdHorizontal.tsx:64]: nit: Consider creating variables for the triangle's width and height to make this a bit easier to read

File: packages/ui-components/src/components/MetricLegendThreshold/utils.ts [hunk 6]
```diff
@@ -10,9 +10,13 @@ export interface CategoryItem {
   description: string | undefined;
   /** Formatted value of the threshold at the end of the current category */
   endThreshold?: string;
+  showIndicator: boolean;
```
Comment (pnavarrc) [inline packages/ui-components/src/components/MetricLegendThreshold/utils.ts:14]: Consider adding the docstring to it shows up in the docs

**Review by pnavarrc — APPROVED** [review]
Looks good! Added a couple of minor suggestions. Remember to add the changeset!

File: packages/ui-components/src/components/LegendThreshold/LegendThreshold.style.ts (hunk 4 above)
Comment (chasulin) [reply packages/ui-components/src/components/LegendThreshold/LegendThreshold.style.ts:41]: the added 1px makes it a tinyyy bit larger

**Pull request merged**